### PR TITLE
DataSource, Driver and ResultSet

### PR DESCRIPTION
**Pull request by fedefernandez**
This PR adds datasource, driver and result set metatada implementations

Please @juanpedromoreno could you review? Thanks

### CODE REVIEW

File: src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDataSource.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  def getConnection(username: String, password: String): Connection = getConnection
+
+  def getLogWriter: PrintWriter =
+    Try {
+      new PrintWriter("droid.log")
+    } match {
+      case Success(pw) => pw
+      case Failure(e) =>
+        e.printStackTrace()
```
Comment (juanpedromoreno) [inline src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDataSource.scala]: Should be wrapped into the log?

File: src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDataSource.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        javaNull
+    }
+
+  def setLogWriter(out: PrintWriter) =
+    Try {
+      DriverManager.setLogWriter(out)
+    } match {
+      case Failure(e) => e.printStackTrace()
```
Comment (juanpedromoreno) [inline src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDataSource.scala]: Should be wrapped into the log?

File: src/main/scala/com/fortysevendeg/android/sqlite/LogWrapper.scala [hunk 3]
```diff
@@ -1,40 +1,66 @@
...
+  def i(message: String)
 
-    def e(message: String)
+  def w(message: String)
 
-    def e(message: String, t: Throwable)
+  def e(message: String)
 
-    def i(message: String)
+  def e(message: String, t: Throwable)
```
Comment (juanpedromoreno) [inline src/main/scala/com/fortysevendeg/android/sqlite/LogWrapper.scala]: does it make sense if we re-define `e` method in this way?

def e(message: String, t: Option[Throwable] = None)? IMO, this Option could be justified :P

File: src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDataSource.scala [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  dbPath: String,
+  log: LogWrapper = new AndroidLogWrapper) extends DataSource {
+
+  private[this] val url: String = "jdbc:sqldroid:" + dbPath
+
+  lazy val connection: Connection = driver.connect(url, properties)
+
+  def getConnection: Connection = connection
+
+  def getConnection(username: String, password: String): Connection = getConnection
```
Comment (juanpedromoreno) [inline src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDataSource.scala]: We are not using username and password, can we remove this method?

File: src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDriver.scala [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.logging.Logger
+import SQLDroidDriver._
+import org.sqldroid.SQLDroidConnection
+
+import scala.util.{Failure, Success, Try}
+
+class SQLDroidDriver extends Driver {
+
+  override def acceptsURL(url: String): Boolean =
+    url.startsWith(sqlDroidPrefix) || url.startsWith(sqlitePrefix)
```
Comment (juanpedromoreno) [inline src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDriver.scala]: In Java world.. this url can be a null?

File: src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDataSource.scala (hunk 4 above)
Comment (fedefernandez) [inline src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDataSource.scala]: Sorry, I forget to add the `override` qualification

File: src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDriver.scala (hunk 5 above)
Comment (fedefernandez) [inline src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidDriver.scala]: Good point

**Comment by fedefernandez** [comment]
Comments addressed @juanpedromoreno

**Comment by juanpedromoreno** [comment]
LGTM!

**Pull request merged**